components/chart/editModule/index.tsx

```diff
@@ -92,6 +97,20 @@ const EditDrawer: React.FC<PropsWithChildren<DrawerProps>> = ({ open, lang, onNo
     }
   }, [data, targetModel]);
 
+  useEffect(() => {
+    const loadTranslations = async () => {
+      const fetchedTranslations = await fetchTranslations(lang.Name);
+      setTranslations(fetchedTranslations);
+    };
+    loadTranslations();
+  }, [lang.Name]);
```

Comment:
@Ranim882 why are you fetching the translations here and you are not doing that inside getServerSideProps inside pages/index as we discussed, was there a problem implementing it there and passing it to TemplateRouter as prop?
They were suggesting that inside the article https://nextjsstarter.com/blog/nextjs-multilingual-ssr-localization-guide/

Also why you removed translations from the ThemeProvider?